### PR TITLE
Add cv32e40p support for pulpissimo, pulp and control_pulp

### PR DESCRIPTION
**Pull request by alex96295**
# **Overview**

Extend #21 #22 with cv32e40p support in control_pulp. This pr cherry-picks some commits from [cv32_target](https://github.com/pulp-platform/pulp-runtime/tree/cv32_target).
It may replace the latter for cv32 support in pulp-runtime on pulp-systems different than `control_pulp` (`pulpissimo`, `pulp-open`)

# **Checklist**

- [x] Add cv32e40p support for control pulp

### CODE REVIEW

**Comment by alex96295** [comment]
@bluewww rfc when you have time

It would be great to know if this is what you had in mind, or if we need to narrow it down more for control_pulp

File: rules/pulpos/targets/pulpissimo.mk [hunk 1]
```diff
@@ -4,12 +4,18 @@ PULP_CFLAGS += -D__ibex__ -U__riscv__ -UARCHI_CORE_HAS_PULPV2 -DRV_ISA_RV32
...
+PULP_LDFLAGS += 
+PULP_CFLAGS += -D__cv32e40p__ -U__riscv__ -UARCHI_CORE_HAS_PULPV2 
+PULP_ARCH_CFLAGS ?=  -march=rv32imcxgap9 
+PULP_ARCH_LDFLAGS ?=  -march=rv32imcxgap9 
+PULP_ARCH_OBJDFLAGS ?= -Mmarch=rv32imcxgap9 
 else
 PULP_LDFLAGS      += 
 PULP_CFLAGS       +=  -D__riscv__
-PULP_ARCH_CFLAGS ?=  -march=rv32imcxgap9
+PULP_ARCH_CFLAGS ?=  -march=rv32imcxgap9
```
Comment (bluewww) [inline rules/pulpos/targets/pulpissimo.mk:16]: trailing spaces

**Review by bluewww — COMMENTED** [review]
where do we define `__cv32e40p__` when selecing the control-pulp target?

File: rules/pulpos/targets/pulpissimo.mk (hunk 1 above)
Comment (alex96295) [reply rules/pulpos/targets/pulpissimo.mk:16]: Thanks

**Comment by alex96295** [comment]
It is defined under `rules/pulpos/target/control-pulp.mk`

**Comment by alex96295** [comment]
@bluewww I tried this out in control_pulp, it has no regression. ack?

**Comment by bluewww** [comment]
Ok I squashed the useless commits and improved the error messages, but otherwise I didn't touch anything

**Review by bluewww — APPROVED** [review]
lgtm

**Pull request merged**